src/k8s/cmd/util/hooks.go

```diff
@@ -0,0 +1,57 @@
+package cmdutil
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+)
+
+// getFileOwnerAndGroup retrieves the UID and GID of a file.
+func getFileOwnerAndGroup(filePath string) (uid, gid uint32, err error) {
+	// Get file info using os.Stat
+	fileInfo, err := os.Stat(filePath)
+	if err != nil {
+		return 0, 0, fmt.Errorf("error getting file info: %w", err)
+	}
+	// Convert the fileInfo.Sys() to syscall.Stat_t to access UID and GID
+	stat, ok := fileInfo.Sys().(*syscall.Stat_t)
+	if !ok {
+		return 0, 0, fmt.Errorf("failed to cast to syscall.Stat_t")
+	}
+	// Return the UID and GID
+	return stat.Uid, stat.Gid, nil
+}
+
+// ValidateRootOwnership checks if given path owner root and root group.
+func ValidateRootOwnership(path string) (err error) {
+	UID, GID, err := getFileOwnerAndGroup(path)
+	if err != nil {
+		return err
+	}
+	if UID != 0 {
+		return fmt.Errorf("owner of %s is user with UID %d expected 0", path, UID)
+	}
+	if GID != 0 {
+		return fmt.Errorf("owner of %s is group with GID %d expected 0", path, GID)
+	}
+	return nil
+}
+
+// InLXDContainer checks if k8s runs in lxd container.
+func InLXDContainer() (isLXD bool, err error) {
+
+	initialProcessEnvironmentVariables := "/proc/1/environ"
```

Comment:
this should be `initialProcessEnvironmentVariable` (singular) as already noted previously.

Reply:
it depends, this file keeps [initial environment of the program](https://man7.org/linux/man-pages/man5/proc_pid_environ.5.html) and in that particular case holds only one variable. So do we want to assume we are in lxd env if so I can change it, but in a generic context it should be plural IMHO.